kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/ConfigurationServiceImpl.java

```diff
@@ -1032,21 +1057,17 @@ private void writeSnapshot(long sid, XmlComponentConfigurations conf) throws Kur
         } catch (IOException e) {
             throw new KuraIOException(e);
         }
+    }
 
+    private File getTempSnapshotFile(File fSnapshot) throws KuraIOException {
+        File tempSnapshotFile;
         try {
-            // Consolidate snapshot writing
-            Files.move(tempSnapshotFile.toPath(), fSnapshot.toPath(), StandardCopyOption.REPLACE_EXISTING,
-                    StandardCopyOption.ATOMIC_MOVE);
-
-        } catch (IOException e) {
-            try {
-                Files.delete(tempSnapshotFile.toPath());
-            } catch (IOException e1) {
-                throw new KuraIOException(e1);
-            }
-            throw new KuraIOException(e);
+            tempSnapshotFile = File.createTempFile(fSnapshot.getName(), null, new File(fSnapshot.getParent()));
+            tempSnapshotFile.deleteOnExit();
```

Comment:
Is this really necessary? it will add a new exit hook for each snapshot leaking memory. Is it better to delete the temporary file if it exists in a finally block instead?

Reply:
I think we always had it: https://github.com/eclipse-kura/kura/blob/5c2f432e10e1feef54ec37c134de08094f24951d/kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/ConfigurationServiceImpl.java#L1014
I have just moved it around to refactor in separate methods to improve readability.

I'll have a look based on your suggestion